### PR TITLE
Created matcher to compare xmls

### PR DESCRIPTION
**Pull request by olenagerasimova**
Working on #272 I've discovered that xmlunit does not work with `Path` instances - it simply always matches. To fix this problem I've created `IsXmlEqual` matcher and used it where necessary.

### CODE REVIEW

File: src/main/java/com/artipie/rpm/hm/IsXmlEqual.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    public boolean matchesSafely(final Path item) {
+        return CompareMatcher
+            .isIdenticalTo(item.toFile())
+            .ignoreWhitespace().normalizeWhitespace().matches(this.xml);
+    }
+
+    @Override
+    @SuppressWarnings("PMD.StringInstantiation")
+    public void describeTo(final Description description) {
+        description.appendText(new String(this.xml));
```
Comment (olegmoz) [inline src/main/java/com/artipie/rpm/hm/IsXmlEqual.java:79]: @olenagerasimova conversion of bytes into `String` might fail, as generally not every byte array can be converted to `String`. Though this scenario is unlikely, so I am not sure this really need fixing. Maybe if we specify `US_ASCII` encoding here we avoid exceptions in any case

File: src/main/java/com/artipie/rpm/hm/IsXmlEqual.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.cactoos.scalar.Unchecked;
+import org.hamcrest.Description;
+import org.hamcrest.TypeSafeMatcher;
+import org.xmlunit.matchers.CompareMatcher;
+
+/**
+ * Matcher for xmls.
```
Comment (olegmoz) [inline src/main/java/com/artipie/rpm/hm/IsXmlEqual.java:34]: @olenagerasimova guess `xmls` should be capitalized

**Comment by 0crat** [comment]
This pull request [#274](https://github.com/artipie/rpm-adapter/issues/274) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/97f5db94-1102-49f2-bbec-a3e5ca152fec); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by olenagerasimova** [comment]
@olegmoz thanks, corrected, approve please

**Comment by 0crat** [comment]
Job was finished in 0 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vatavuk[/z](https://www.0crat.com/u/vatavuk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@0crat quality bad

**Comment by 0crat** [comment]
Quality is low, no payment, see [§31](http://www.zerocracy.com/policy.html#31)

**Pull request merged**